lib/atc/aws/remote_fixity_check.rb

```diff
@@ -95,11 +95,17 @@ def welcome_message?(data)
   end
 
   def confirm_subscription_message?(data, job_identifier)
-    data['type'] == 'confirm_subscription' && JSON.parse(data['identifier'])['job_identifier'] == job_identifier
+    return false if data['identifier'].nil?
+
+    data['type'] == 'confirm_subscription' && JSON.parse(data['identifier'])&.fetch('job_identifier') == job_identifier
   end
 
   def custom_message?(data, job_identifier)
-    data['type'].nil? && JSON.parse(data['identifier'])['job_identifier'] == job_identifier && data['message'].present?
+    return false if data['type'].present?
+    return false if data['message'].nil?
```

Comment:
Should 'blank?' be used instead of 'nil?' on data['message']? The replaced code uses present?, and the converse is blank?, I believe. This could make a difference for an empty string, where "".blank? returns true (and "".present? returns false), but "".nil? returns false.
I'm trying to map the logic of the new code to the code it replaces, but it is possible the logic changed.

Reply:
I'm expecting any non-custom message sent by ActionCable to have a `data['type']` value, and any custom message to completely omit the `type` key, so you've made me realize that line 104 would be better written as `return false unless data['type'].nil?` (because even a `data['type']` value of `""` would indicate that this is NOT a custom message).  Glad you mentioned that, thank you!  I'll update that.

For line 105, all custom messages should have a 'message' key that holds a string value.  Even though it's incredibly unlikely that string value would be an empty string, even an empty string message would still technically count as a valid custom message, so I think that one still makes sense to keep as a `.nil?` check.